src/services/anchor/index.ts

```diff
@@ -105,17 +162,38 @@ export const sep10 = async (
     throw new Error(`Invalid sequence number: ${transactionSigned.sequence}`);
   }
 
-  const { clientSignature, clientPublic } = await fetchSep10Signatures(
-    transactionSigned.toXDR(),
-    outputToken,
-    accountId,
+  let signatureData = await checkAndWaitForSignature();
```

Comment:
We will fetch the signature, if it's invalid or doesn't exist, will ask the user to sign and only then continue.